apps/nextra/pages/en/build/sdks/ts-sdk/type-safe-contract.mdx

```diff
@@ -0,0 +1,130 @@
+---
+title: "Surf: TypeScript Type Safety for Move Contracts"
+---
+
+import { Steps, Tabs } from 'nextra/components'
+
+
+# Surf: TypeScript Type Safety for Move Contracts
+
+## What is Surf
+
+Surf is a TypeScript library built on top of the Aptos TypeScript SDK and the wallet adapter that provides static type safety for your Move contracts by inferring type from contract ABI (Application Binary Interface). It allows you to catch type errors at compile time rather than at runtime. Most existing TypeScript IDEs will automatically provide warnings if you try to access fields that don't exist, or provide wrong input types.
+
+## Usage
+
+<Steps>
+
+### Step 1
+
+First, download the ABI of the Move contract and save it to a TypeScript file. In this case, we're naming the file `abi.ts` in the `src/utils` folder.
+
+<Tabs items={['Mac & Linux', 'Windoes']}>
+  <Tabs.Tab>
+    ```bash filename="get_abi.sh"
+    #! /bin/bash
+
+    # replace it with the network your contract lives on
+    NETWORK=testnet
+    # replace it with your contract address
+    CONTRACT_ADDRESS=0x12345
+    # replace it with your module name, every .move file except move script has module_address::module_name {}
+    MODULE_NAME=fungible_asset_launchpad
+
+    # save the ABI to a TypeScript file
+    echo "export const ABI = $(curl https://fullnode.$NETWORK.aptoslabs.com/v1/accounts/$CONTRACT_ADDRESS/module/$MODULE_NAME | sed -n 's/.*"abi":\({.*}\).*}$/\1/p') as const" > src/utils/abi.ts
+    ```
+  </Tabs.Tab>
+  <Tabs.Tab>
+    ```json filename="package.json"
+    {
+      "name": "my-aptos-dapp",
+      "version": "0.0.0",
+      "scripts": {
+        "get-abi": "node gen_abi.js"
+      },
+      "dependencies": {
+        "axios": "^1.7.2",
+        "js-yaml": "^4.1.0"
+      }
+    }
+    ```
+
+    ```js filename="gen_abi.js"
+    const axios = require("axios");
+    const fs = require("node:fs");
+
+    // replace it with the network your contract lives on
+    NETWORK=testnet
+    // replace it with your contract address
+    CONTRACT_ADDRESS=0x12345
+    // replace it with your module name, every .move file except move script has module_address::module_name {}
+    MODULE_NAME=fungible_asset_launchpad
+
+    async function getAbi() {
+      axios
+        .get(`https://fullnode.${process.env.VITE_APP_NETWORK}.aptoslabs.com/v1/accounts/0x${accountAddress}/module/${MODULE_NAME}`)
+        .then((response) => {
+          const abi = response.data.abi;
+          const abiString = `export const ABI = ${JSON.stringify(abi)} as const;`;
+          // Write ABI to abi.ts file
+          fs.writeFileSync("src/utils/abi.ts", abiString);
+          console.log("ABI saved to src/utils/abi.ts");
+        })
+        .catch((error) => {
+          console.error("Error fetching ABI:", error);
+        });
+    }
+
+    getAbi();
+    ```
+
+    ```PowerShell filename="get_abi.ps1"
+    npm i 
+    npm run get-abi
+    ```
+  </Tabs.Tab>
```

Comment:
I can unblock this for now.

However, please replace this with a call in create-aptos-dapp, because right now, you have 2 scripts that do the same thing.  Ideally, it should only be the second one.